packages/account/src/Sections/Security/ConnectedApps/connected-apps.tsx

```diff
@@ -1,163 +1,101 @@
-                    <DataList.Cell row={row} column={columns_map.app_id} is_footer={!is_appstore} />
-                </div>
-            </div>
-        ),
-        [columns_map, is_appstore]
-    );
-
-    const revokeConnectedApp = React.useCallback(async (app_id: number | null) => {
-        setLoading(true);
-        const response = await WS.authorized.send({ revoke_oauth_app: app_id });
-        if (!response.error) {
-            /* eslint-disable no-console */
-            fetchConnectedApps().catch(error => console.error('error: ', error));
-        } else {
-            setError(true);
-        }
-    }, []);
-
     const handleRevokeAccess = React.useCallback(() => {
-        setModalVisibility(false);
-        revokeConnectedApp(selected_app_id);
-    }, [revokeConnectedApp, selected_app_id]);
+        if (selected_app_id) revokeOAuthApp(selected_app_id);
+        setIsModalOpen(false);
+    }, [revokeOAuthApp, selected_app_id]);
 
-    return (
-        <section
-            className={classNames('connected-apps__wrapper', {
-                'connected-apps__wrapper--dashboard': is_appstore,
-            })}
-        >
-            <Text color='prominent' weight='bold' as='p' className='connected-apps__title'>
-                {localize('Authorised applications')}
-            </Text>
-            {is_error && <ErrorComponent />}
-            <div
-                className={classNames('connected-apps__container', 'connected-apps__wrapper', {
-                    'connected-apps__wrapper--dashboard': is_appstore,
-                })}
-            >
-                {is_loading ? (
-                    <Loading is_fullscreen={false} />
+    return is_loading ? (
+        <Loading is_fullscreen={false} />
+    ) : (
+        <div className='connected-apps__wrapper'>
+            <section>
+                {is_error && <ErrorComponent />}
+                {connected_apps.length ? (
+                    <div className='connected-apps__content--wrapper'>
+                        <ConnectedAppsInfo />
+                        {is_mobile ? (
+                            <div className='connected-apps__list--wrapper'>
+                                {connected_apps.map(connected_app => (
+                                    <DataListTemplate
+                                        key={connected_app.app_id}
+                                        data_source={connected_app}
+                                        handleToggleModal={handleToggleModal}
+                                    />
+                                ))}
+                            </div>
+                        ) : (
+                            <div className='connected-apps__tabular--wrapper'>
+                                <DataTable
+                                    className='connected-apps'
+                                    data_source={connected_apps as unknown as TSource[]}
```

Comment:
looks like `connected_apps` is coming from `useOAuthConnectedApps and it doesn't has the correct types 
Instead of type casting, can we update the types from the hook or the API?

Reply:
DataTable is a shared component. so touching it is not a good idea, at the moment. ☠️